R/modify-model-field.R

```diff
@@ -46,6 +47,11 @@ modify_model_field <- function(.mod, .field, .value, .append = TRUE, .remove = F
     checkmate::assert_character(.value, null.ok = TRUE)
   }
 
+  if (isTRUE(.bool_value)) {
+    .value <- unlist(.value)
+    checkmate::assertLogical(.value, null.ok = TRUE)
```

Comment:
is NULL okay?

Reply:
Do you have any thoughts what the default should be? Do you think it makes more sense for it to be NULL or FALSE? I was thinking it if was set to NULL, the user would know they had not reviewed the model yet.

Reply:
> Do you have any thoughts what the default should be?

It's not clear to me that there is a scenario where a `NULL` would reach `modify_model_field`.

As you have it now, everything that has a `.starred = NULL` default (`copy_model_from`, `copy_model_from.bbi_nonmem_model`, `copy_nonmem_model_from`) ultimately goes through `new_model`, and that only sends through values that are not `NULL`.

https://github.com/metrumresearchgroup/bbr/blob/74baeb15d44eab79fdd110fb81626fbea9212b50/R/new-model.R#L72

So, I think `NULL` makes sense as a default "leave it be" value for `new_model` and friends, but I don't see why `modify_model_field` needs to be aware of that (or allow it with `null.ok = TRUE`).

This is somewhat related to [discussion elsewhere in this PR](https://github.com/metrumresearchgroup/bbr/pull/487#discussion_r880627005), but here's the (exported) interface that seems most natural to me:

```
copy_model_from(..., .starred = NULL)
copy_model_from.bbi_nonmem_model(..., .starred = NULL)
new_model(..., .starred = NULL)

add_star <- function(.mod)
remove_star <- function(.mod)
```

Reply:
To come back to this, I guess the state I'm describing is that the user can only set the star to `FALSE` or `TRUE`, but not return it to its unset factory default (`NULL`).  That feels okay to me both `mod$starred` of `NULL` or `FALSE` both mean the model is unstarred, so I don't see why the caller would care _how_ we unstar it.